### PR TITLE
[HOLD] Updated tslint to use api based approach.

### PR DESCRIPTION
**Pull request by Blackbaud-BobbyEarl**
Issue: https://github.com/blackbaud/skyux-sdk-builder/issues/52

**Updates**

Also addresses: https://github.com/blackbaud/skyux-sdk-builder/issues/50

Documented: https://github.com/blackbaud/skyux2-docs/pull/332

### CODE REVIEW

File: cli/test.js [hunk 1]
```diff
@@ -31,12 +31,11 @@ function test(command, argv) {
...
 
   const onRunComplete = () => {
+    // Print linting errors again after coverage reporter
     if (lintResult && lintResult.exitCode > 0) {
-      // Pull the logger out of the execution stream to let it print
-      // after karma's coverage reporter.
       setTimeout(() => {
-        logger.error('Process failed due to linting errors:');
-        lintResult.errors.forEach(error => logger.error(error));
+        logger.error('TSLlint errors displayed again for convenience:');
```
Comment (Blackbaud-SteveBrush) [inline cli/test.js]: "TSLlint"

File: cli/utils/ts-linter.js [hunk 2]
```diff
@@ -1,51 +1,62 @@
...
 
-const flags = [
-  '--project',
-  skyPagesConfigUtil.spaPath('tsconfig.json'),
-  '--config',
-  skyPagesConfigUtil.spaPath('tslint.json'),
-  '--exclude',
-  '**/node_modules/**/*.ts'
-];
+const { Configuration, Linter, Formatters } = tslint;
```
Comment (Blackbaud-SteveBrush) [inline cli/utils/ts-linter.js]: Just my personal preference, but do you need the `tslint` variable anywhere? Could this be imported as:
```
const { Configuration, Linter, Formatters } = require('tslint);
```

File: cli/test.js (hunk 1 above)
Comment (Blackbaud-BobbyEarl) [reply cli/test.js]: Fixed.

File: cli/utils/ts-linter.js (hunk 2 above)
Comment (Blackbaud-BobbyEarl) [reply cli/utils/ts-linter.js]: Adjusted.

**Comment by Blackbaud-BobbyEarl** [comment]
Ready for another look @Blackbaud-SteveBrush.  I addressed the comments you had, including the initializing problem (so that you can change `fit`/`fdescribe` during `skyux watch`).

File: cli/utils/ts-linter.js [hunk 3]
```diff
@@ -1,51 +1,85 @@
...
+  const suffix = arr.length === 1 ? '' : 's';
+  return `${arr.length} ${word}${suffix}`;
+}
+
+function getOptions(argv) {
+  const options = {
+    formatter: 'stylish'
+  };
+
+  if (argv.fix) {
```
Comment (Blackbaud-SteveBrush) [inline cli/utils/ts-linter.js]: I believe this line is causing the `e2e` build step to fail.

**Comment by Blackbaud-BobbyEarl** [comment]
Closing in favor https://github.com/blackbaud/skyux-sdk-builder/pull/57

**Pull request closed without merging**